pkg/provider/cloud/openstack/helper.go

```diff
@@ -88,9 +88,9 @@ func getNetworkByName(netClient *gophercloud.ServiceClient, name string, isExter
 	}
 
 	candidates := []*NetworkWithExternalExt{}
-	for _, n := range existingNetworks {
+	for i, n := range existingNetworks {
 		if n.External == isExternal {
-			candidates = append(candidates, &n)
+			candidates = append(candidates, &existingNetworks[i])
```

Comment:
oh wow, this one was dangerous!

Reply:
Not really. `candidates` is only used if its length is 1, and if its length is 1, the loop had only one iteration, so no pointer overwriting could have occured. That was also the only reason this ever worked :D

Reply:
... so yeah, in that sense the old code was dangerous. ^^